client/state/query-client-ssr.ts

```diff
@@ -1,13 +1,52 @@
-import { QueryClient, QueryCache } from 'react-query';
+import { QueryClient, QueryCache, dehydrate, QueryOptions, QueryKey } from 'react-query';
 import { MAX_AGE, BASE_STALE_TIME } from 'calypso/state/initial-state';
+import type { Query, QueryState } from 'react-query/types/core/query';
 
-const sharedCache = new QueryCache();
+class QueryClientSSR extends QueryClient {
+	private fetchedQueryKeys: { [ hash: string ]: boolean } = {};
```

Comment:
This could be a `Set`, with `.add` and `.has` method.

Then `getFetchedQueryKey` always returns `boolean`, never `undefined`, and its calls don't need to use `!!`.

Reply:
Thank you! It's much cleaner now.